### PR TITLE
Add ChaincodeData and CDSData protos

### PR DESCRIPTION
**Pull request by GuillaumeCisco**
Adding CDSData as defined https://github.com/hyperledger/fabric/blob/master/core/common/ccprovider/cdspackage.go#L36
And ChaincodeData as defined https://github.com/hyperledger/fabric/blob/master/core/common/ccprovider/ccprovider.go#L431

### CODE REVIEW

File: peer/chaincode.proto [hunk 1]
```diff
@@ -85,3 +87,44 @@ message ChaincodeInvocationSpec {
 message LifecycleEvent {
     string chaincode_name = 1;
 }
+
+// https://github.com/hyperledger/fabric/blob/master/core/common/ccprovider/cdspackage.go#L36
```
Comment (sykesm) [inline peer/chaincode.proto]: Please remove the ilnks. They're unstable and unnecessary in the context of the message documentation.

File: peer/chaincode.proto [hunk 2]
```diff
@@ -85,3 +87,44 @@ message ChaincodeInvocationSpec {
...
+//CDSData is data stored in the LSCC on instantiation of a CC
```
Comment (sykesm) [inline peer/chaincode.proto]: Please use [slash][slash][space] for the doc comments. I understand that we don't do that in all comments but we're working towards that.

**Comment by sykesm** [comment]
@GuillaumeCisco all contributions require a DCO signoff. The details link from the DCO bot will explain what you need to do.

File: peer/chaincode.proto [hunk 3]
```diff
@@ -85,3 +87,44 @@ message ChaincodeInvocationSpec {
...
+//hence the protobuf format
+message CDSData {
+    bytes hash = 1; // hash of ChaincodeDeploymentSpec.code_package
+    bytes metadatahash = 2; // hash of ChaincodeID.name + ChaincodeID.version
+}
+
+// https://github.com/hyperledger/fabric/blob/master/core/common/ccprovider/ccprovider.go#L431
+// ChaincodeData defines the datastructure for chaincodes to be serialized by proto
+// Type provides an additional check by directing to use a specific package after instantiation
+// Data is Type specifc (see CDSPackage and SignedCDSPackage)
```
Comment (jyellick) [inline peer/chaincode.proto]: specific

**Comment by GuillaumeCisco** [comment]
> @GuillaumeCisco all contributions require a DCO signoff. The details link from the DCO bot will explain what you need to do.

Just did it ;)

File: peer/chaincode.proto (hunk 3 above)
Comment (GuillaumeCisco) [reply peer/chaincode.proto]: thanks

File: peer/chaincode.proto (hunk 2 above)
Comment (GuillaumeCisco) [reply peer/chaincode.proto]: done

File: peer/chaincode.proto (hunk 1 above)
Comment (GuillaumeCisco) [reply peer/chaincode.proto]: done

**Comment by GuillaumeCisco** [comment]
> As mentioned in Gerrit, overall this seems fine to me (pending addressing Matt's comments). For posterity, I will say it one more time, these protos support the old (now deprecated) `lscc` style lifecycle invocations. The newer `_lifecycle` returns messages which are not opaque and are already available via the peer/lifecycle set of protos.

I agree, but the 1.* versions are stable release, so I think we could still add some improvements :)

**Pull request merged**